herders/filters.py

```diff
@@ -111,29 +125,49 @@ def filter_monster__skills__skill_effect__pk(self, queryset, name, value):
             # Filter effects based on effects of each individual skill. This ensures a monster will not show up unless it has
             # the desired effects on the same skill rather than across any skills.
             skills = Skill.objects.all()
+            skills_count = skills.count()
 
             for effect in value:
                 skills = skills.filter(skill_effect=effect)
 
             for pk in stat_scaling:
                 skills = skills.filter(scaling_stats=pk)
 
-            cooltime_filter = Q(cooltime__gte=min_cooltime) & Q(cooltime__lte=max_cooltime)
+            cooltime_filter = Q()
+
+            if max_cooltime is not None and max_cooltime > 0:
+                cooltime_filter &= Q(cooltime__lte=max_cooltime)
+            
+            if min_cooltime is not None and min_cooltime > 0:
+                cooltime_filter &= Q(cooltime__gte=min_cooltime)
+
             if min_cooltime == 0 or max_cooltime == 0:
-                cooltime_filter = Q(cooltime__isnull=True) | cooltime_filter
+                cooltime_filter |= Q(cooltime__isnull=True)
 
-            skills = skills.filter(
-                cooltime_filter,
-                hits__lte=max_num_hits,
-                hits__gte=min_num_hits,
-            )
+            if cooltime_filter:
+                skills = skills.filter(cooltime_filter)
+            
+            hits_filter = Q()
+
+            if max_num_hits:
+                hits_filter &= Q(hits__lte=max_num_hits)
+
+            if min_num_hits:
+                hits_filter &= Q(hits__gte=min_num_hits)
+
+            if hits_filter:
+                skills = skills.filter(hits_filter)
 
             if passive is not None:
                 skills = skills.filter(passive=passive)
 
             if aoe is not None:
                 skills = skills.filter(aoe=aoe)
 
+            # no skill filters
+            if skills_count == skills.count():
```

Comment:
Just noting that same answer from monster filter optimization question applies here